src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2416,21 +2446,28 @@ public ConstructorEdit(INamedTypeSymbol oldType)
                                     }
                                     else if (oldSymbol.DeclaringSyntaxReferences.Length == 1 && newSymbol.DeclaringSyntaxReferences.Length == 1)
                                     {
+                                        // Handles partial methods and record members
+
                                         // We ignore partial method definition parts when processing edits (GetSymbolForEdit).
                                         // The only declaration in compilation without syntax errors that can have multiple declaring references is a type declaration.
                                         // We can therefore ignore any symbols that have more than one declaration.
                                         ReportTypeLayoutUpdateRudeEdits(diagnostics, newSymbol, edit.NewNode, newModel, ref lazyLayoutAttribute);
 
-                                        // Compare the old declaration syntax of the symbol with its new declaration and report rude edits
-                                        // if it changed in any way that's not allowed.
                                         var oldNode = GetSymbolDeclarationSyntax(oldSymbol.DeclaringSyntaxReferences[0], cancellationToken);
                                         var newNode = edit.NewNode;
 
-                                        ReportDeclarationInsertDeleteRudeEdits(diagnostics, oldNode, newNode);
+                                        // Compare the old declaration syntax of the symbol with its new declaration and report rude edits
+                                        // if it changed in any way that's not allowed.
+                                        ReportDeclarationInsertDeleteRudeEdits(diagnostics, oldNode, newNode, oldSymbol, newSymbol);
+
+                                        // If a node has been inserted but neither old nor new has a body, we can stop processing
+                                        // unless its explicitly implenenting a positional property of a record, in which case we
+                                        // still need to issue constructor edits below.
+                                        var isRecordPrimaryConstructorParameter = IsRecordPrimaryConstructorParameter(oldNode);
```

Comment:
>IsRecordPrimaryConstructorParameter [](start = 82, length = 35)

I'm not sure I understand this case - in what scenarios do we get an insert of a primary constructor parameter here? Changing primary consturctor parameters is a rude edit, right?

Is this connected to the change on line 2420? Where the property declaration itself is implicit but DeclaringSyntaxReferences returns the primary constructor parameter syntax?

Reply:
> I'm not sure I understand this case - in what scenarios do we get an insert of a primary constructor parameter here?

Not an insert of a primary constructor parameter, but an explicit implementation of a property that is backed by one.

What this code used to do is essentially "The thing being added has no initializer, so no need to do anything else". That doesn't hold true for explicitly implemented properties, where the property represents a positional record parameter. In those cases, the compiler generated property did have an initializer so even if the user doesn't specify one in their explicit implementation, it still constitutes a change in the initializer, so we need a constructor edit.

Having said that, I will revisit this code (and try to improve the wording of the comment) in line with your other comments about the `if (oldSymbol.IsImplicitlyDeclared)` line. I was trying to avoid duplicating too much logic, but I'm not against it if its easier to read and understand :)

Reply:
I updated the comment, the rest of the changes here are required, nothing to do with the `IsImplicitlyDeclared` case